plugins/wallabag.koplugin/main.lua

```diff
@@ -315,13 +334,62 @@ function Wallabag:getBearerToken()
     end
 end
 
+--- Get a JSON formatted list of articles from the server.
+-- The list should have self.article_per_sync item, or less if an error occured.
+-- If filter_tag is set, only articles containing this tag are queried.
+-- If ignore_tags is defined, articles containing either of the tags are skipped.
 function Wallabag:getArticleList()
     local filtering = ""
+    local ignoring = {}
     if self.filter_tag ~= "" then
         filtering = "&tags=" .. self.filter_tag
     end
-    local articles_url = "/api/entries.json?archive=0&perPage=" .. self.articles_per_sync .. filtering
-    return self:callAPI("GET", articles_url, nil, "", "")
+    if self.ignore_tags ~= "" then
+        local tag
```

Comment:
No need to define this; the for loop should be local.
```suggestion
```

```
Luacheck results
Checking plugins/wallabag.koplugin/main.lua       4 warnings

    plugins/wallabag.koplugin/main.lua:348:15: unused variable 'tag'
    plugins/wallabag.koplugin/main.lua:350:13: shadowing definition of variable 'tag' on line 348
    plugins/wallabag.koplugin/main.lua:371:19: unused variable 'tag'
    plugins/wallabag.koplugin/main.lua:373:20: shadowing definition of variable 'tag' on line 371
```

Reply:
Yep I was going to fix this this evening, I didn't know that this variable was local to the loop.
Should've understood this from the other iterators. Good thing there is CI :)